src/distributors/adoptopenjdk-installer.ts

```diff
@@ -0,0 +1,83 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+
+import { IS_WINDOWS, PLATFORM, IS_MACOS, macOSJavaContentDir } from "../util";
+import { IJavaInfo, IJavaRelease, JavaBase, JavaInstallerOptions } from "./base-installer";
+import { IRelease, IReleaseVersion } from './adoptopenjdk-models'
+
+export class AdoptOpenJdkDistributor extends JavaBase {
+    
+    constructor(initOptions: JavaInstallerOptions) {
+        super("AdoptOpenJDK", initOptions);
+    }
+
+    private async getAvailableMajor(range: semver.Range) {
+        const urlReleaseVersion = "https://api.adoptopenjdk.net/v3/info/available_releases"
+        const javaVersionAvailable = (await this.http.getJson<IReleaseVersion>(urlReleaseVersion)).result;
+
+        if (!javaVersionAvailable) {
+            throw new Error(`No versions were found for ${this.distributor}`)
+        }
+
+        const javaSemVer = javaVersionAvailable.available_releases.map(item => semver.coerce(item)!)!;
+        const majorVersion = semver.maxSatisfying(javaSemVer, range)?.major;
+
+        if(!majorVersion) {
+            throw new Error(`Could find version which satisfying. Versions: ${javaVersionAvailable.available_releases}`);
+        }
+
+        return majorVersion;
+    }
+
+    protected async downloadTool(javaRelease: IJavaRelease): Promise<IJavaInfo> {
+        let toolPath: string;
+        let downloadDir: string;
+
+        core.info(`Downloading ${this.distributor}, java version ${javaRelease.resolvedVersion}`);
+        const javaPath = await tc.downloadTool(javaRelease.link);
+        
+        if(IS_WINDOWS) {
+            downloadDir = await tc.extractZip(javaPath);
+        } else {
+            downloadDir = await tc.extractTar(javaPath);
+        }
+
+        const archiveName = fs.readdirSync(downloadDir)[0];
+        const archivePath = path.join(downloadDir, archiveName);
+        toolPath = await tc.cacheDir(archivePath, `Java_${this.distributor}_${this.javaPackage}`, javaRelease.resolvedVersion, this.arch);
+
+        if(process.platform === 'darwin') {
+            toolPath = path.join(toolPath, macOSJavaContentDir);
+        }
+
+        return { javaPath: toolPath, javaVersion: javaRelease.resolvedVersion };
+    }
+
+    protected get javaRootName(): string {
```

Comment:
toolcacheFolderName

Reply:
please moved this function to base class

Reply:
Done.